src/cron-job.ts

```diff
@@ -0,0 +1,251 @@
+import { ApiObject, CronOptions, Duration, Lazy, Schedule } from 'cdk8s';
+import { Construct } from 'constructs';
+import * as timezone from 'moment-timezone';
+import * as k8s from './imports/k8s';
+import { JobProps } from './job';
+import { RestartPolicy } from './pod';
+import * as workload from './workload';
+
+/**
+ * Concurrency policy for CronJobs.
+ */
+export enum ConcurrencyPolicy {
+  ALLOW = 'Allow',
+  FORBID = 'Forbid',
+  REPLACE = 'Replace',
+}
+
+/**
+ * Properties for `CronJob`.
+ */
+export interface CronJobProps extends workload.WorkloadProps {
+  /**
+   * Properties of the Job that is being scheduled by CronJob.
+   */
+  readonly jobProperties: JobProps;
+
+  /**
+   * Specifies the time in which the job would run again.
+   *
+   * @default - Runs at every minute.
+   */
+  readonly schedule?: CronOptions;
+
+  /**
+   * Specifies the timezone for the job. This helps aligining the schedule to follow the specified timezone.
+   *
+   * @default - Timezone of kube-controller-manager process.
+   */
+  readonly timeZone?: string;
+
+  /**
+   * Specifies the concurrency policy for the job.
+   *
+   * @default - Allow
+   */
+  readonly concurrencyPolicy?: ConcurrencyPolicy;
+
+  /**
+   * Specifies the time by which a recurring job should be scheduled after it has passed its expected scheduling time.
+   * Missed jobs are counted as failed ones.
+   *
+   * @default - There is no deadline.
+   */
+  readonly startingDeadline?: Duration;
+
+  /**
+   * Specifies if the recurring job should be suspended.
+   *
+   * @default - false.
+   */
+  readonly suspend?: boolean;
+
+  /**
+   * Specifies the number of successful jobs history retained.
+   *
+   * @default - 3.
+   */
+  readonly successfulJobsRetained?: number;
+
+  /**
+   * Specifies the number of failed jobs history retained.
+   *
+   * @default - 1.
+   */
+  readonly failedJobsRetained?: number;
+}
+
+/**
+ * A CronJob is responsible for creating a Job and scheduling it based on provided scheduled. This helps running Jobs in a recurring manner.
+ */
+export class CronJob extends workload.Workload {
+  /**
+   * Properties of the recurring `Job`.
+   */
+  public readonly jobProperties: JobProps;
+
+  /**
+   * Schedule for which to run the recurring job. By default, the job would be scheduled to run every minute.
```

Comment:
These are now no longer properties, they are attributes / fields. That is, they should describe the behavior of a would be instance, but the behavior of the current one.

```suggestion
   * The schedule this cron job is scheduled to run in.
```

Please rephrase all of these to convey the correct context.

Reply:
I see, will update it.